requirements-docs.txt

```diff
@@ -0,0 +1,5 @@
+furo==2023.07.26
+myst-parser==2.0.0
+sphinx==6.2.1
+
+-r requirements.txt
```

Comment:
Out of curiosity, does putting these documentation requirements in `requirements-dev.txt` and using that in the `.readthedocs.yaml` file cause issues? Also, if we do keep this separate, does it actually need to pull in `requirements.txt` ?

Reply:
> Out of curiosity, does putting these documentation requirements in requirements-dev.txt and using that in the .readthedocs.yaml file cause issues?

No, I initially was actually using `requirements-build.txt`. I separated this into a new file because there's no need to install sphinx and associated packages for testing (Main reason to install `requirements-dev.txt`) or creating frozen builds (Main reason to install `requirements-build.txt`) so it would seem wasteful not to separate out those dependencies for the relatively rare case that someone wants to build documentation locally.

> Also, if we do keep this separate, does it actually need to pull in requirements.txt ?

Currently, no. I included it because I'm under the impression that if we use some sort of auto-documentation that generates documentation by evaluating our actual python code, that we would need our required dependencies installed. But since I'm not positive of that and it's trivial to add back later I'll remove it for now.